packages/arb-token-bridge-ui/src/hooks/useNetworksAndSigners.tsx

```diff
@@ -180,17 +185,31 @@ export function NetworksAndSignersProvider(
     }
   }, [isTosAccepted, openConnectModal])
 
+  const invocationCounter = useRef(0)
```

Comment:
The `invocationCounter` logic adds way too much complexity to an already complex business logic in this hook. Could we figure out a way to fix the issue it fixes, using an alt approach? Or in the worst case, add comments about its presence?

Reply:
Yeah it's a weird solution, but I could make anything else work without restructuring the component. What happens here the hook can be triggered twice when L1 <> L2 <> Orbit combination is chosen and needs to rearrange itself

Some iterations run faster and need to make sure only the latest is grabbed. Would love a different solution and I'm open to ideas, but nothing else would work